crates/amaru-sim/src/bin/amaru-sim/simulator.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
I think this module shouldn't be called "simulator". A simulator is spawning several nodes and handles the message passing between them. This module is essentially the main function of one of those nodes. Perhaps  a better name is "gasket_eventloop/pipeline"?

Reply:
Or `node` ?